client/app/services/query.js

```diff
@@ -43,6 +43,43 @@ class QueryResultError {
 }
 
 
+class Parameter {
+  constructor(parameter) {
+    this.title = parameter.title;
+    this.name = parameter.name;
+    this.type = parameter.type;
+    this.value = parameter.value;
+    this.global = parameter.global;
+  }
+
+  get ngModel() {
+    if (this.type === 'date' || this.type === 'datetime-local' || this.type === 'datetime-with-seconds') {
+      this.$$value = this.$$value || moment(this.value).toDate();
+      return this.$$value;
+    } else if (this.type === 'number') {
+      this.$$value = this.$$value || parseInt(this.value, 10);
+      return this.$$value;
+    }
+
+    return this.value;
+  }
+
+  set ngModel(value) {
+    if (value && this.type === 'date') {
+      this.value = moment(value).format('YYYY-MM-DD');
+      this.$$value = moment(this.value).toDate();
+    } else if (value && this.type === 'datetime-local') {
+      this.value = moment(value).format('YYYY-MM-DDTHH:mm');
```

Comment:
BigQuery sadly doesn't support this format (it doesn't like the `T`) so we won't be able to use this (we've been using the fix in #1553) until it's configurable.